packages/cloudfront-request-presigner/src/sign.ts

```diff
@@ -0,0 +1,221 @@
+import { parseUrl } from "@aws-sdk/url-parser";
+import { createSign } from "crypto";
+import { readFileSync } from "fs";
+
+interface PolicyDates {
```

Comment:
please rearrange the declarations in the file:

exported interfaces: move to the top under imports, add comments for what the fields are

exported functions: move to the top under exported interfaces, add jsdoc comments if only to point to the readme.

non-exported interfaces: if the interface plays a role in the user _input_ of the exported functions, also export that interface. Otherwise you can leave them under exported functions, or under exported interfaces.

non-exported functions: place below exported functions